### ISSUE TITLE
Allow adding multiple Axes/Header for the same channel

### ISSUE THREAD

**Issue by kanitw** [vega/vega-lite#214]
Make `encoding.axis` supports an array of axis

**Comment by kanitw** [comment]
Not sure if auto-adding is a good idea, but having this as a special option might be useful in some cases.   I'm moving this to "Should have later" though.

**Comment by domoritz** [comment]
Please elaborate what this means.

**Comment by kanitw** [comment]
> Make `encoding.axis` supports an array of axis

(instead of one axis) -- e.g., just in cases uses want axis on both left and right of the plot.

**Comment by kanitw** [comment]
Another alternative is to make `orient` support `"both"` or an array of multiple values. 

We can also have finer grain control by having domainOrient, tickOrient, etc. too. 

Similar for header.

**Comment by kanitw** [comment]
I start to think that instead of supporting array of axes for each encoding,
it might be better to make each axis property (except values) support array for repeating. 

For example, `orient: ['left', 'right']` would mean "both" (and "both" could be a macro for `['left', 'right']`
 
If there are multiple arrays, then treat them as parallel array.  For example,

```
{
  orient: ['left', 'right']
  labels: true,
  title:  [true, false]
}
```

would mean showing title on the left only

**Comment by kanitw** [comment]
This way, the required declaration would be less redundant.

**Comment by domoritz** [comment]
I don't agree. If I read the spec, I'd think that all exes should labeled. If laves was an array worth one value, I'd think that only the left is labeled.

**Comment by kanitw** [comment]
I'm not sure what you're talking about.  

> I'd think that all exes should labeled

All axes would be labeled with the spec above (`true` = `[true, true]`), but it would only include title on the left.
